File: google/drive/commands/delete_permission.py
import os
from apis.permissions import delete_permission
from apis.helper import get_client


def delete_permission_tool() -> None:
    client = get_client("drive", "v3")
    file_id = os.getenv("FILE_ID")
    permission_id = os.getenv("PERMISSION_ID")

    if not file_id:
        print("Error: FILE_ID environment variable is required but not set")
        return
    if not permission_id:
        print("Error: PERMISSION_ID environment variable is required but not set")
        return

    success = delete_permission(client, file_id, permission_id)
    if success:
        print(f"Successfully deleted permission: {permission_id}")
    else:
        print(f"Failed to delete permission: {permission_id}")

File: microsoft365/onedrive/pkg/commands/deletePermission.go
package commands

import (
	"context"
	"fmt"

	"github.com/obot-platform/tools/microsoft365/onedrive/pkg/client"
	"github.com/obot-platform/tools/microsoft365/onedrive/pkg/global"
	"github.com/obot-platform/tools/microsoft365/onedrive/pkg/graph"
)

func DeletePermission(ctx context.Context, driveID string, itemID string, permissionID string) error {
	if driveID == "me" {
		return fmt.Errorf("drive_id must be the actual drive ID, cannot be 'me'")
	}

	c, err := client.NewClient(global.AllScopes)
	if err != nil {
		return fmt.Errorf("failed to create client: %w", err)
	}

	err = graph.DeleteDriveItemPermission(ctx, c, driveID, itemID, permissionID)
	if err != nil {
		return fmt.Errorf("failed to delete permissions: %w", err)
	}

	fmt.Printf("Successfully deleted permission for item %s in drive %s\n", itemID, driveID)
	return nil
}
